Assets/Scripts/Layers/LayerUI.cs

```diff
@@ -690,5 +690,16 @@ public void OnPointerExit(PointerEventData eventData)
             if (!IsSelected)
                 SetHighlight(InteractionState.Default);
         }
+
+        private void OnDestroy()
+        {
+            if(IsSelected)
```

Comment:
Hoort de Logica hieronder niet in de LayerData zelf thuis? Met bijvoorbeeld een methode `RemoveLayer`? Daarmee behoudt je encapsulatie en hoeft deze class geen toegang te hebben tot interne staat van de LayerData

Reply:
Dit heb ik juist hier, omdat het anders via de layerManager gaat, een dit is juist gewoon een cleanup van zichzelf in de LayerData